### DISCUSSION TITLE
New soundfont?

### DISCUSSION THREAD

**Discussion by alejandro6626** [helge17/tuxguitar#97] [General]
I'd like to suggest a change in the default Soundfont.

I typically use GeneralUser GS 1.471 SoundFont from (https://schristiancollins.com/generaluser.php) which I think has good drum and bass sound so that you can play guitar on top with a lot better sound. Also other sounds are very decent and it's half the size of the Magic Soundfont that comes with TuxGuitar.

It has an free and open license, so I don't see an issue in incorporating it into TG.

**Comment by Sy256** [comment]
Wow sounds great from that demo I'd be all for it

**Comment by helge17** [comment]
Thanks for the tip! The GeneralUser soundfont sounds good, quite similar and maybe even better than the current MagicSFver2, so it could be an alternative for future versions. But the soundfont discussion is complex and has many aspects: Personal taste, technical aspects like size and compatibility, and most importantly the licensing issue.

I have tried to contact the author of MagicSFver2 by mail, but have not received a reply. Most or all Linux distributions ship TuxGuitar without MagicSFver2 because of the unclear license. I also asked debian-legal if MagicSFver2 can be shipped with TuxGuitar, but the (short) answer was no; for the log version see https://lists.debian.org/debian-legal/2023/01/msg00004.html and follow-up.

Therefore the Debian package for TuxGuitar 1.5.6 currently uses the soundfont FluidR3_GM from the fluid-soundfont-gm package.

GeneralUser is currently included in the Debian package minuet-data, but the problem is the same as with MagicSFver2: nobody can guarantee that it is 100% free, see Debian bug report https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1036719 and the links to the debian-legal mailing list therein.

MagicSFver2 has been the default soundfont in TuxGuitar for many years, so I think there must be a good reason to change this. The licensing situation of GeneralUser is not that much better than MagicSFver2. The size is better/smaller, I’m not sure about the rest.

On Android, TuxGuitar again uses a different soundfont.

Maybe it would be best not to ship a soundfont with the TuxGuitar program itself, but to provide some kind of download manager where users can choose which soundfont they want to download and use. Looking at the source code, it seems that something like this already existed: E.g. "jsa.soundbank-assistant" in the message files and TuxGuitar-jsa/src/org/herac/tuxguitar/player/impl/jsa/assistant/SBAssistant.java. The file contains links to some soundfonts provided by Oracle. Maybe the code could be ractivated and extended. But this is just an idea, we should think about it more carefully before making further efforts.

**Comment by alejandro6626** [comment]
Agree with what you say. I actually use Fluidsynth with that Soundfont, so I don't care much about Tuxguitar coming prepacked with any soundfont. However more newbie users that want an out-of-the-box experience surely appreciate Tuxguitar having an embedded soundfont. I think it's a great feature.

SF sounds are subjective, yes, however I have tested dozens of Soundfonts, and I truly think that is one of the best out there. I use it for creating backing tracks and playing over them, I specially like the sound of drums.

License seems to be ok to include it

*** GeneralUser GS v1.471 ***
***      License v2.0     ***

** License of the complete work **
You may use GeneralUser GS without restriction for your own music creation, private or commercial.  This SoundFont bank is provided to the community free of charge.  Please feel free to use it in your software projects, and to modify the SoundFont bank or its packaging to suit your needs.
